clients/java/src/main/java/io/camunda/client/impl/search/request/UsersByGroupSearchRequestImpl.java

```diff
@@ -55,50 +54,52 @@ public UsersByGroupSearchRequestImpl(
     this.jsonMapper = jsonMapper;
     this.groupId = groupId;
     httpRequestConfig = httpClient.newRequestConfig();
-    request = new UserSearchQueryRequest();
+    request = new GroupUserSearchQueryRequest();
   }
 
   @Override
-  public FinalSearchRequestStep<User> requestTimeout(final Duration requestTimeout) {
+  public FinalSearchRequestStep<GroupUser> requestTimeout(final Duration requestTimeout) {
     httpRequestConfig.setResponseTimeout(requestTimeout.toMillis(), TimeUnit.MILLISECONDS);
     return this;
   }
 
   @Override
-  public CamundaFuture<SearchResponse<User>> send() {
+  public CamundaFuture<SearchResponse<GroupUser>> send() {
     ArgumentUtil.ensureNotNullNorEmpty("groupId", groupId);
-    final HttpCamundaFuture<SearchResponse<User>> result = new HttpCamundaFuture<>();
+    final HttpCamundaFuture<SearchResponse<GroupUser>> result = new HttpCamundaFuture<>();
     httpClient.post(
         String.format("/groups/%s/users/search", groupId),
         jsonMapper.toJson(request),
         httpRequestConfig.build(),
-        UserSearchResult.class,
-        SearchResponseMapper::toUsersResponse,
+        GroupUserSearchResult.class,
+        SearchResponseMapper::toGroupUsersResponse,
         result);
     return result;
   }
 
   @Override
-  public UsersByGroupSearchRequest filter(final UserFilter value) {
-    request.setFilter(provideSearchRequestProperty(value));
+  public UsersByGroupSearchRequest filter(final GroupUserFilter value) {
+    // This command doesn't support filtering
```

Comment:
:question: What about throwing `UnsupportedOperationException` instead?

Reply:
It feels a bit stupid to provide users with this method and then have it be unsupported and actaully throw an exception 😅 This way when a user tries to it has to use the filter builder, which doesn't provide any methods to implement. I felt that was better, but it's both not ideal tbh.

Reply:
> In that case, we could even forward the filter all the way to where we actually issue the query instead of dropping it here.

Not sure what you mean by this

Reply:
Currently the `UsersByGroupSearchRequest` just drops the filter while `SearchQueryRequestMapper#toGroupQuery` just creates a new and empty `GroupFilter`.
To future-proof this a bit more, we could at least forward the filter from the request until we get to `SearchQueryRequestMapper#toGroupQuery` where we then do nothing because `GroupUserFilter` is empty.

But it's just a nitpick, feel free to ignore.